src/Faker/Calculator/Ean.php

```diff
@@ -20,7 +20,7 @@ class Ean
      * @param string $digits
      * @return int
      */
-    public static function checksum($digits)
+    public static function checksum(string $digits): int
```

Comment:
One can consider this a BC break. 

The class is not final and if someone extended this class and override this function, then they will get incompatible types. 

https://3v4l.org/jTe3N

Reply:
Did not think about that to be honest. Only other problem it actual means that everything we do on the lib is almost breaking. What are your thoughts on that?

Reply:
Yeah. Keeping BC could be tricky of this library. 

I think we should do 2 things. 
1. Mark things with deprecation errors everywhere we **need to**. Using [this package](https://github.com/symfony/deprecation-contracts) is helpful. 
2. Add a BC checker so we dont accidentally do something strange.